src/Controller/Entry/EntryFrontController.php

```diff
@@ -17,46 +16,62 @@
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\Security\Http\Attribute\IsGranted;
 
 class EntryFrontController extends AbstractController
 {
     public function __construct(private readonly EntryRepository $repository)
     {
     }
 
-    public function root(?string $sortBy, ?string $time, ?string $type, Request $request): Response
+    public function root(?string $sortBy, ?string $time, ?string $type, string $federation, Request $request): Response
     {
-        $user = $this->getUser();
-
-        if (!$user) {
-            return $this->front($sortBy, $time, $type, $request);
-        }
-
-        $front = match ($user->homepage) {
-            User::HOMEPAGE_SUB => 'subscribed',
-            User::HOMEPAGE_MOD => 'moderated',
-            User::HOMEPAGE_FAV => 'favourite',
-            default => 'front',
-        };
-
-        return $this->$front($sortBy, $time, $type, $request);
+        return $this->front($sortBy, $time, $type, $request->query->get('filter'), $federation, $request);
     }
 
-    public function front(?string $sortBy, ?string $time, ?string $type, Request $request): Response
+    public function front(?string $sortBy,
+        ?string $time,
+        ?string $type,
+        ?string $filter,
+        string $federation,
+        Request $request): Response
     {
```

Comment:
I'd prefer this to be single line. I don't know what our guidelines are on this though

Reply:
Done